src/types/deleted.rs

```diff
@@ -39,7 +41,16 @@ use std::ops::RangeInclusive;
 /// # }
 /// ```
 #[derive(Debug, Clone)]
-pub enum Deleted {
+pub struct Deleted {
```

Comment:
Let's mark this as `#[non_exhaustive]` in case we realize there are other expunge details we want to add in the future.